modules/ref_fb_module/tests/test_fb_statistics.cpp

```diff
@@ -0,0 +1,428 @@
+
+                        // Assert that domain sample end equals expected value
+                        ASSERT_EQ(domainData->high, expectedDomainEnd[i][ii]);
+                    }
+                    else
+                    {
+                        // In case of Implicit or Explicit
+                        auto domainData = static_cast<Int*>(domainPacket.getData());
+                        const size_t domainSampleCount = domainPacket.getSampleCount();
+                        auto domainDataSample = domainData[ii];
+
+                        // Assert that domain packet has expected number of samples
+                        ASSERT_EQ(domainSampleCount, expectedDomain[i].size());
+
+                        // Assert that domain sample equals expected value
+                        ASSERT_EQ(domainDataSample, expectedDomain[i][ii]);
+                    }
+                }
+            }
+        }
+    }
+
+    void receivePacketsAndCheckBoth()
+    {
+        receivePacketsAndCheck(expectedAvg, readerAvg);
+        receivePacketsAndCheck(expectedRms, readerRms);
+    }
+};
+
+TEST_F(StatisticsTest, StatisticsTestBasic)
+{
+    vecvec<Float> mockPackets{{0.1, 0.2, 0.3, 0.4, 0.5, 0.6, 0.7, 0.8, 0.9, 1.0},
+                              {1.1, 1.2, 1.3, 1.4, 1.5, 1.6, 1.7, 1.8, 1.9, 2.0},
+                              {2.1, 2.2, 2.3, 2.4, 2.5, 2.6, 2.7, 2.8, 2.9, 3.0}};
+    vecvec<Float> expectedAvg{{0.55}, {1.55}, {2.55}};
+    vecvec<Float> expectedRms{{0.62048368229954287}, {1.5763882770434448}, {2.5661254840712679}};
+    vecvec<Int> expectedDomain{{3}, {23}, {43}};
+    Int initialOutputDomain = 0;  // Implicit
+
+    auto helper = StatisticsTestHelper(LinearDataRule(2, 3),
+                                       initialOutputDomain,
+                                       expectedAvg,
+                                       expectedRms,
+                                       expectedDomain,
+                                       SampleTypeFromType<Float>::SampleType,
+                                       mockPackets);
+    helper.run();
+}
+
+TEST_F(StatisticsTest, StatisticsTestSmallerPackets)
+{
+    vecvec<Float> mockPackets{{0.1, 0.2, 0.3},
+                              {0.4, 0.5, 0.6},
+                              {0.7, 0.8, 0.9},
+                              {1.0, 1.1, 1.2},
+                              {1.3, 1.4, 1.5},
+                              {1.6, 1.7, 1.8},
+                              {1.9, 2.0, 2.1},
+                              {2.2, 2.3, 2.4},
+                              {2.5, 2.6, 2.7},
+                              {2.8, 2.9, 3.0}};
+    vecvec<Float> expectedAvg{{}, {}, {}, {0.55}, {}, {}, {1.55}, {}, {}, {2.55}};
+    vecvec<Float> expectedRms{{}, {}, {}, {0.62048368229954287}, {}, {}, {1.5763882770434448}, {}, {}, {2.5661254840712679}};
+    vecvec<Int> expectedDomain{{}, {}, {}, {3}, {}, {}, {23}, {}, {}, {43}};
```

Comment:
Are we expecting the statistics fb to output empty packets here?

Reply:
We are not expecting any packets back at that point. It is up to the function block implementation to store input packet values in the buffer unit the BlockSize is reached, which in this example is after the fourth, seventh and tenth input packets.